homeassistant/components/recorder/__init__.py

```diff
@@ -386,6 +413,9 @@ def async_purge(now):
                 if not self.entity_filter(entity_id):
                     continue
 
+            if not self.enabled:
+                continue
+
```

Comment:
Put this before `if event.event_type in self.exclude_t:` since there is no need to check the filter or excludes for events we won't be recording.